src/processor/operator/recursive_extend/frontier_scanner.cpp

```diff
@@ -51,10 +51,15 @@ bool PathScanner::trailSemanticCheck(const std::vector<nodeID_t>&,
     const std::vector<relID_t>& edgeIDs) {
     common::rel_id_set_t set;
     for (auto i = 0u; i < edgeIDs.size() - 1; ++i) {
-        if (set.contains(edgeIDs[i])) {
+        internalID_t edgeID = edgeIDs[i];
+        if (RelIDMasker::needFlip(const_cast<internalID_t&>(edgeIDs[i]))) {
```

Comment:
Good catch. This is indeed a bug.

I would suggest some minor coding convention changes.

```
if (RelIDMasker::needFlip(const_cast<internalID_t&>(edgeIDs[i])))
```
The `const_cast` doesn't seem to be needed. You can probably just do
```
if (RelIDMasker::needFlip(edgeID))
```